rustfmt.toml

```diff
@@ -0,0 +1,7 @@
+use_small_heuristics = "Max"
+use_field_init_shorthand = true
+use_try_shorthand = true
+empty_item_single_line = true
+edition = "2018"
+unstable_features = true
+fn_args_density = "Compressed"
```

Comment:
do you have a rationale somewhere for moving away from the defaults ?

Reply:
If I remember correctly, mostly because of `fn_args_density`; without it, rustfmt will often format this:

```rust
func(x1, x2, x3, x4, x5, x6, x7, x8, x9)
```

as 

```rust
func(
    x1,
    x2,
    x3,
    x4,
    x5,
    x6,
    x7,
    x8,
    x9,
)
```

even though its still within line length limit.

Reply:
all right, I'll give it a try on the rest of the codebase and see if i like it :)